src/switch_rtp.c

```diff
@@ -5752,6 +5812,18 @@ SWITCH_DECLARE(void) switch_rtp_destroy(switch_rtp_t **rtp_session)
 		switch_core_timer_destroy(&(*rtp_session)->write_timer);
 	}
 
+	if ((*rtp_session)->fork.fork_rx.transcoding) {
+		(*rtp_session)->fork.fork_rx.transcoding = 0;
+		switch_core_codec_destroy(&(*rtp_session)->fork.fork_rx.codec_in);
+		switch_core_codec_destroy(&(*rtp_session)->fork.fork_rx.codec_out);
+	}
+
+	if ((*rtp_session)->fork.fork_tx.transcoding) {
```

Comment:
This doesn't leak, it will be gone with session. Also PCM codecs (it is design to work with them only at the moment) do not allocate anything, they are basically single routine to check value from array of values.
It is fixed in amended commit anyway.